packages/dd-trace/src/appsec/sdk/track_event.js

```diff
@@ -63,7 +69,112 @@ function trackCustomEvent (tracer, eventName, metadata) {
 
   trackEvent(eventName, metadata, 'trackCustomEvent', getRootSpan(tracer))
 
-  incrementSdkEventMetric('custom')
+  incrementSdkEventMetric('custom', 'v1')
+
+  if (eventName === 'users.login.success' || eventName === 'users.login.failure') {
+    runWaf(eventName)
+  }
+}
+
+function trackUserLoginSuccessV2 (tracer, login, user, metadata) {
+  if (!login || typeof login !== 'string') {
+    log.warn('[ASM] Invalid login provided to v2.trackUserLoginSuccess')
+    return
+  }
+
+  const rootSpan = getRootSpan(tracer)
+  if (!rootSpan) {
+    log.warn('[ASM] Root span not available in v2.trackUserLoginSuccess')
+    return
+  }
+
+  const wafData = { login }
+
+  metadata = {
+    'usr.login': login,
+    ...metadata
+  }
+
+  if (user) {
+    if (typeof user !== 'object') {
+      user = { id: user }
+    }
+
+    if (user.id) {
+      wafData.id = user.id
+      setUserTags(user, rootSpan)
+      metadata.usr = user
+    }
+  }
+
+  trackEvent('users.login.success', metadata, 'v2.trackUserLoginSuccess', rootSpan)
+
+  runWaf('users.login.success', wafData)
+
+  incrementSdkEventMetric('login_success', 'v2')
+}
+
+function trackUserLoginFailureV2 (tracer, login, exists, metadata) {
+  if (!login || typeof login !== 'string') {
+    log.warn('[ASM] Invalid login provided to v2.trackUserLoginFailure')
+    return
+  }
+
+  const rootSpan = getRootSpan(tracer)
+  if (!rootSpan) {
+    log.warn('[ASM] Root span not available in v2.trackUserLoginFailure')
+    return
+  }
+
+  const wafData = { login }
+
+  if (typeof exists === 'object' && typeof metadata === 'undefined') {
```

Comment:
what's the reason for using typeof undefined instead of this ?
```suggestion
  if (typeof exists === 'object' && metadata === undefined) {
```